common/unified/matrix/dense_kernels.template.cpp

```diff
@@ -76,14 +76,22 @@ void scale(std::shared_ptr<const DefaultExecutor> exec,
         run_kernel(
             exec,
             [] GKO_KERNEL(auto row, auto col, auto alpha, auto x) {
-                x(row, col) *= alpha[col];
```

Comment:
I'm wondering if changes like this have a performance impact. I'm a bit skeptical that the compiler is capable of hoisting the inner if-check out of the hot loop. But I also don't want to introduce another if nesting on top.

Reply:
Since this operates column-wise, I don't see an immediate optimization potential. On the other hand, this can be implemented without an actual branch with a CMOV on x86 or something similar on GPUs to avoid the branch overhead. Should even be possible to vectorize with mask registers, but I'm not sure if the auto-vectorizer can already do that.

Reply:
No longer relevant, I removed this change